src/main/java/net/syscon/elite/repository/jpa/model/BedAssignmentHistory.java

```diff
@@ -0,0 +1,68 @@
+package net.syscon.elite.repository.jpa.model;
+
+import lombok.AllArgsConstructor;
+import lombok.Builder;
+import lombok.Data;
+import lombok.EqualsAndHashCode;
+import lombok.NoArgsConstructor;
+import lombok.ToString;
+
+import javax.persistence.Column;
+import javax.persistence.Embeddable;
+import javax.persistence.EmbeddedId;
+import javax.persistence.Entity;
+import javax.persistence.JoinColumn;
+import javax.persistence.ManyToOne;
+import javax.persistence.Table;
+import java.io.Serializable;
+import java.time.LocalDate;
+import java.time.LocalDateTime;
+
+@Data
+@Builder
+@EqualsAndHashCode(exclude = {"bedAssignmentHistoryPK", "offenderBooking"}, callSuper = false)
+@ToString(exclude = {"bedAssignmentHistoryPK", "offenderBooking"})
+@NoArgsConstructor
+@AllArgsConstructor
+@Entity
+@Table(name = "BED_ASSIGNMENT_HISTORIES")
+public class BedAssignmentHistory extends AuditableEntity {
+
+    @Data
+    @NoArgsConstructor
+    @AllArgsConstructor
+    @Embeddable
+    public static class BedAssignmentHistoryPK implements Serializable {
+        @Column(name = "OFFENDER_BOOK_ID", nullable = false)
+        private Long offenderBookingId;
+
+        @Column(name = "BED_ASSIGN_SEQ", nullable = false)
+        private Integer sequence;
+    }
+
+    @EmbeddedId
+    private BedAssignmentHistoryPK bedAssignmentHistoryPK;
+
+    @ManyToOne
+    @JoinColumn(name = "OFFENDER_BOOK_ID", insertable = false, updatable = false)
+    private OffenderBooking offenderBooking;
+
+    @Column(name = "LIVING_UNIT_ID")
+    private Long livingUnitId;
+
+    @Column(name = "ASSIGNMENT_DATE")
+    private LocalDate assignmentDate;
+
+    @Column(name = "ASSIGNMENT_TIME")
+    private LocalDateTime assignmentDateTime;
+
+    @Column(name = "ASSIGNMENT_REASON")
+    private String assignmentReason;
+
+    @Column(name = "ASSIGNMENT_END_DATE")
+    private LocalDate assignmentEndDate;
+
+    @Column(name = "ASSIGNMENT_END_TIME")
+    private LocalDateTime assignmentEndDateTime;
```

Comment:
should this be LocalTime ?

Reply:
It's a strange thing, but the table has 2 columns which are both timestamps, but one only shows the date and the other shows both the date and time.  I'm not 100% sure what the above will do on an Oracle database so thanks for reminding me to check that out before I start using it on t3.

Reply:
Its a terrible way that its implemented - something to do with the way dates use to work in Oracle 5/6